lib/osv/ecosystems.py

```diff
@@ -310,23 +310,74 @@ def enumerate_versions(self, package, introduced, fixed, limits=None):
     return self._get_affected_versions(versions, introduced, fixed, limits)
 
 
+class Debian(Ecosystem):
+  """Debian ecosystem"""
+
+  _API_PACKAGE_URL = 'https://snapshot.debian.org/mr/package/{package}/'
+  versions_to_idx = None
+  debian_release_ver: str
+
+  def __init__(self, debian_release_ver: str):
+    self.debian_release_ver = debian_release_ver
+    debian_version_cache.initiate_from_cloud_cache()
+
+  def sort_key(self, version):
+    if self.versions_to_idx is None:
+      raise Exception('versions to idx not initiated')
+
+    try:
+      return self.versions_to_idx[version]
+    except KeyError:
+      raise EnumerateError(f'Version {version} does not exist') from KeyError
+
+  def enumerate_versions(self, package, introduced, fixed, limits=None):
+    url = self._API_PACKAGE_URL.format(package=package.lower())
+    response = requests.get(url)
+    if response.status_code == 404:
+      raise EnumerateError(f'Package {package} not found')
+    if response.status_code != 200:
+      raise RuntimeError(
+          f'Failed to get Debian versions for {package} with: {response.text}')
+
+    response = response.json()
+
+    versions = [x['version'] for x in response['result']]
+    # Reverse so versions is in ascending order
+    versions.reverse()
+    self.versions_to_idx = dict(zip(versions, range(len(versions))))
+
+    if introduced == '0':
+      # Update introduced to the first version of the debian version
+      introduced = debian_version_cache.get_first_package_version(
+          package, self.debian_release_ver)
+
+    return self._get_affected_versions(versions, introduced, fixed, limits)
+
+
+# All lowercase ecosystem names
```

Comment:
technical ecosystem names are case sensitive. could we revert this? it doesn't make our matching any easier right?

Reply:
Will revert this. Is Debian's ecosystem name capitalized? i.e. `Debian:9` or `debian:9`?

Reply:
Using capitalized version